### PR TITLE
Feature/94 clean up code in task after audio testing

### PR DESCRIPTION
**Pull request by ICTEnablers**
No need to merge this before the sprint demo

### CODE REVIEW

File: src/components/task/task.tsx [hunk 1]
```diff
@@ -98,63 +98,50 @@ const Task = props => {
...
-      wrongAudioElement.current &&
-      !['Control', 'Meta', 'Shift', 'Alt'].some((modifier: string): boolean => event.key === modifier)
-    ) {
-      audioElement.current.pause();
-      audioElement.current.setAttribute('src', '');
-      audioElement.current = new Audio();
-      correctAudioElement.current.load();
-      wrongAudioElement.current.load();
-
+      audioElement.current) {
```
Comment (duff92) [inline src/components/task/task.tsx]: Place this on one line :)

File: src/components/task/task.tsx [hunk 2]
```diff
@@ -98,63 +98,50 @@ const Task = props => {
...
       if (currentPos + 1 === task.exercise.length && correctKeyPressed) {
         completedAction(task);
         props.history.push('/summary');
       }
 
-      correctKeyPressed ? handleCorrectInputAction(event.key).then(() => {
-        if (correctAudioElement.current) {
-          correctAudioElement.current.setAttribute('currentTime', '0');
-          correctAudioElement.current.play().then(() => {
+      if(correctKeyPressed) {
```
Comment (duff92) [inline src/components/task/task.tsx]: Space missing after if

File: src/components/task/task.tsx [hunk 3]
```diff
@@ -98,63 +98,50 @@ const Task = props => {
...
-        wrongAudioElement.current!.setAttribute('currentTime', '0');
-        wrongAudioElement.current!.play().then(() => {
+              if (textURL !== '' && audioElement.current) {
+                playAudio(audioElement, textURL).catch(error => console.error('playAudio error ', error));
+              }
+            }).catch(error => console.error('speak error', error));
+          }
+        }).catch(error => console.error('play current error', error));
+      }
+      else {
```
Comment (duff92) [inline src/components/task/task.tsx]: `else` should be placed on same line

```
} else {
```

File: src/components/task/task.tsx [hunk 4]
```diff
@@ -98,63 +98,50 @@ const Task = props => {
     if (inputElement && inputElement.current) {
       inputElement.current.focus();
     }
-    speak(task.exercise[currentPos].text, ttsOptions).then(url => playAudio(audioElement, url));
   }, [currentPos, task.exercise, ttsOptions]);
 
+  useEffect(() => {
+    speak(task.exercise[currentPos].text, ttsOptions).then(url => playAudio(audioElement, url));
```
Comment (duff92) [inline src/components/task/task.tsx]: Add catch for `playAudio` and `speak`

File: src/components/audio/audio.tsx [hunk 5]
```diff
@@ -3,15 +3,16 @@ export const playAudio = async (audio, url: string, rate = 1): Promise<string> =
     return Promise.reject(`playAudio URL: ${url} audio: ${audio}`);
   } else {
     return new Promise((resolve, reject) => {
-      audio.current.playbackRate = rate;
-      audio.current.src = url;
       audio.current.pause();
+      audio.current.setAttribute('src','');
+      
+      audio.current.setAttribute('playbackRate', rate);
+      audio.current.setAttribute('src', url)  ;
```
Comment (duff92) [inline src/components/audio/audio.tsx]: Why are you first setting `src` two lines before and on this line as well?

File: src/components/audio/audio.tsx [hunk 6]
```diff
@@ -3,15 +3,16 @@ export const playAudio = async (audio, url: string, rate = 1): Promise<string> =
...
       audio.current.load();
       audio.current.addEventListener('ended', resolve);
       audio.current.addEventListener('error', reject);
-
-      const promise = audio.current.play();
-      if (promise !== undefined) {
+      const promise = audio.current.play().catch(error => console.error('playAudio error '+error));
```
Comment (duff92) [inline src/components/audio/audio.tsx]: Add spaces in string `console.error('playAudio error ' + error)`

File: src/components/task/task.tsx (hunk 1 above)
Comment (ICTEnablers) [reply src/components/task/task.tsx]: done

File: src/components/task/task.tsx (hunk 2 above)
Comment (ICTEnablers) [reply src/components/task/task.tsx]: fixed

File: src/components/task/task.tsx (hunk 3 above)
Comment (ICTEnablers) [reply src/components/task/task.tsx]: fixed

File: src/components/audio/audio.tsx (hunk 5 above)
Comment (ICTEnablers) [reply src/components/audio/audio.tsx]: the first occurence is supposed to gracefully stop ongoing playback, setting src = ''

File: src/components/audio/audio.tsx (hunk 6 above)
Comment (ICTEnablers) [reply src/components/audio/audio.tsx]: fixed

File: src/components/task/task.tsx (hunk 4 above)
Comment (ICTEnablers) [reply src/components/task/task.tsx]: fixed

**Pull request merged**